domain/dailydebitlimit.go

```diff
@@ -97,6 +97,9 @@ func (r *dailyDebitLimit) wouldExceedLimit(amount int64) bool {
 }
 
 func makeInstanceID(date string, accountID string) string {
+	// validate the date
+	startOfBusinessDay(date)
```

Comment:
I think this should have a separate function specifically for validating - it has nothing to do with "start of business day" really.